src/allmydata/scripts/magic_folder_cli.py

```diff
@@ -545,13 +546,92 @@ def status(options):
     return 0
 
 
+def rawstatus(options):
+    """
+    Dump JSON from the magic-folder status endpoint
+    """
+    nodedir = options["node-directory"]
+    stdout, stderr = options.stdout, options.stderr
+    magic_folders = load_magic_folders(os.path.join(options["node-directory"]))
+
+    with open(os.path.join(nodedir, u'private', u'api_auth_token'), 'rb') as f:
+        token = f.read()
+
+    print("Magic-folder status for '{}':".format(options["name"]), file=stderr)
+
+    if options["name"] not in magic_folders:
+        raise Exception(
+            "No such magic-folder '{}'".format(options["name"])
+        )
+
+    while True:
+        try:
+            magic_data = _get_json_for_fragment(
+                options,
+                'magic_folder?t=json',
+                method='POST',
+                post_args=dict(
+                    t='json',
+                    name=options["name"],
+                    token=token,
+                )
+            )
+        except Exception as e:
+            print("failed to retrieve data: %s" % str(e), file=stderr)
+            return 2
+
+        if isinstance(magic_data, dict) and 'error' in magic_data:
+            print("Error from server: %s" % magic_data['error'], file=stderr)
+            print("This means we can't retrieve the remote shared directory.", file=stderr)
+            return 3
+
+        now = datetime.now()
+        print('[')
+        for st in magic_data:
+            print('  {')
+            for k, v in st.items():
+                print('    "{}": {}'.format(k, v))
+            print('  },')
+        print(']')
```

Comment:
Maybe worth factoring this formatting code into a helper function and directly unit testing it